### ISSUE TITLE
Unstable badge displays old 0.2.0-beta2 release instead of 0.3.x-dev branch

### ISSUE THREAD

**Issue by rob006** [PUGX/badge-poser#342]
Package: https://packagist.org/packages/rob006/flarum-lang-polish

[![Latest Stable Version](https://poser.pugx.org/rob006/flarum-lang-polish/v/stable)](https://packagist.org/packages/rob006/flarum-lang-polish) [![Latest Unstable Version](https://poser.pugx.org/rob006/flarum-lang-polish/v/unstable)](https://packagist.org/packages/rob006/flarum-lang-polish) 

I would expect that unstable badge will show `0.3.x-dev` branch and I'm pretty sure it worked in that way some time ago.

**Comment by rob006** [comment]
This may be related to https://blog.packagist.com/composer-and-default-git-branches/

https://packagist.org/packages/rob006/flarum-lang-polish.json - while branch alias for `dev-master` is set as `0.3.x-dev`, version is not normalized:

```json
"version":"dev-master",
"version_normalized":"dev-master"
```

`0.3.x-dev` is also not listed in list of versions, only `dev-master` is available.

**Comment by JellyBellyDev** [comment]
> This may be related to https://blog.packagist.com/composer-and-default-git-branches/
> 
> https://packagist.org/packages/rob006/flarum-lang-polish.json - while branch alias for `dev-master` is set as `0.3.x-dev`, version is not normalized:
> 
> ```json
> "version":"dev-master",
> "version_normalized":"dev-master"
> ```
> 
> `0.3.x-dev` is also not listed in list of versions, only `dev-master` is available.

@rob006 exactly! :(

**Comment by rob006** [comment]
It works fine for Symfony: ![](https://poser.pugx.org/symfony/symfony/v/unstable), while it is pretty similar situation. So it looks like it does not entirely depends on `version_normalized`. Could this different behavior between my package and Symony be related to `v` prefix in tags names?

**Comment by rob006** [comment]
`dev-master` has now `"default-branch":true` in package JSON - I guess it should be used to internally normalize this branch to `9999999-dev` (this is how Packagist handles it - default branch is always treated as the newest) and keep old behavior for `dev-master` branch.

**Comment by rob006** [comment]
`version_compare()` does not work as expected for versions with `v` prefix: https://3v4l.org/vpJCR
https://github.com/PUGX/badge-poser/blob/308e77170bd51e936f2ef7ed38f9b19e19140066/src/Badge/Model/Package.php#L83

Using [`Comparator` from `composer/semver`](https://github.com/composer/semver/blob/00915994bb1de62e750ae279669c9c5a57379957/src/Comparator.php) may be a better option in this case.

```php
$parser = new VersionParser();
var_dump(Comparator::equalTo($parser->normalize('v0.3.x-dev'), $parser->normalize('0.3.x-dev')))
// bool(true)
```

**Comment by JellyBellyDev** [comment]
@rob006 maybe it's related to this: https://github.com/badges/poser/issues/26#issuecomment-649021711?

**Comment by rob006** [comment]
Yeah, it refers to changes described in the same[ blog post](https://blog.packagist.com/composer-and-default-git-branches/). Previously `dev-master` was normalized to `9999999-dev` (which make it always the latest unstable branch), now it is not normalized at all and `dev-master` is returned as `version_normalized`.

**Comment by rob006** [comment]
Packagist fixed versions ordering on their website, but API response has not been changed: https://github.com/composer/packagist/commit/d313cb83984a0cc2123d91c51d159fd626d51311. I guess that Badge-Poser should use similar solution to handle edge cases like https://packagist.org/packages/friendsofphp/php-cs-fixer correctly.
